plugin_ide.ui/src-lang/melnorme/lang/ide/ui/editor/actions/OpenTypeHandler.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Bruno Medeiros and other Contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Bruno Medeiros - initial API and implementation
+ *******************************************************************************/
+package melnorme.lang.ide.ui.editor.actions;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.dialogs.ElementListSelectionDialog;
+import org.eclipse.ui.texteditor.ITextEditor;
+
+import melnorme.lang.ide.ui.LangUIPlugin_Actual;
+import melnorme.lang.ide.ui.editor.AbstractLangEditor;
+import melnorme.lang.ide.ui.editor.LangEditorMessages;
+import melnorme.lang.ide.ui.editor.structure.EditorStructureUtil;
+import melnorme.lang.ide.ui.utils.operations.BasicEditorOperation;
+import melnorme.lang.tooling.structure.GlobalSourceStructure;
+import melnorme.utilbox.core.CommonException;
+
+public class OpenTypeHandler extends AbstractEditorHandler {
```

Comment:
I am not entirely familiar with the separation of `src` and `src-lang`. The package might be wrong.